pkg/rabbit/connections_handler.go

```diff
@@ -119,23 +118,35 @@ func (r *RabbitMQHelper) SetupRabbitMQConnectionAndChannel(
 	configFunction func(RabbitMQConnectionInterface, RabbitMQChannelInterface) error,
 	dialFunc func(string) (RabbitMQConnectionWrapperInterface, error)) {
 	var err error
-	retryConnection, retryChannel := true, true
-	for retryConnection || retryChannel {
+	for {
 		// Wait one cycle duration always except the first time
 		if !r.firstSetup {
 			time.Sleep(time.Millisecond * r.cycleDuration)
 		} else {
 			r.logger.Info("Creating and Configuring RabbitMQ Connection and Channel")
 			r.firstSetup = false
 		}
-		r.Connection, r.Channel, err = r.CreateAndConfigConnectionsAndChannel(&retryConnection, &retryChannel, rabbitMQURL, configFunction, dialFunc)
-		if err != nil {
-			r.logger.Error(err)
-			if retryConnection {
-				r.CloseRabbitMQConnections()
-			}
+		// create the connection if needed
+		if r.Connection == nil || r.Connection.IsClosed() {
```

Comment:
To keep `SetupRabbitMQConnectionAndChannel` idempotent, we can just call `CloseRabbitMQConnections` at the start of the for loop. That could be the reset to clean up anything leftover.

That function should also set channel to nil as channel is useless without the connection. Also, setting channel to nil is the way to signal to any users of the connection handler that the channel shouldn't be used. So it kind of fits in the "reset" function.

Reply:
the thing here is that im trying not to close channels if the connection is alive. So closing everything makes it idempotent but we loose the capability of reconnecting selectively